tools/ThermoRawFileParser/thermo_converter.xml

```diff
@@ -1,60 +1,227 @@
-<tool id="thermo_raw_file_converter" name="Thermo" version="1.1.2">
+<tool id="thermo_raw_file_converter" name="Thermo" version="1.1.10">
     <description>RAW file converter</description>
     <requirements>
-        <requirement type="package" version="1.1.2">ThermoRawFileParser</requirement>
+        <requirement type="package" version="1.1.10">ThermoRawFileParser</requirement>
     </requirements>
     <command>
 <![CDATA[
+#import re
 
-ln -s '$input' ./input.raw &&
+#set $temp_stderr = "thermo_converter_stderr"
 
-ThermoRawFileParser.sh
-    -i=input.raw
-    -o=./
+echo "" > $temp_stderr &&
+
+mkdir ./raws_folder &&
+mkdir ./output_folder &&
+#for $input_raw in $input:
+    #if len($input) > 1
+        #set $input_name = re.sub('[^\w\-\.]', '_',$input_raw.element_identifier.split('/')[-1].replace(".raw", "") + ".raw")
+        ln -s -f '${input_raw}' './raws_folder/${input_name}' &&
+    #else:
+        ln -s -f '${input_raw}' './raws_folder/input.raw' &&
+    #end if
+#end for
+
+(ThermoRawFileParser.sh
+    -d=./raws_folder
+    -o=./output_folder
     -f=$output_format
-    #if $metadata:
-        --metadata=1
+    #if $gzip_boolean == 1:
+    -g
+    #end if
+    #if $output_metadata_selector != "off":
+        --metadata="${output_metadata_selector}"
+    #end if
+    #if $zlib_boolean == 0:
+    -z
     #end if
+    #if $peakpicking_boolean == 0:
+    -p
+    #end if
+    #if $verbose_boolean == 1:
+    -v
+    #end if
+    #if $ignore_instrument_errors_boolean == 1:
+    -e
+    #end if
+
+    2>> $temp_stderr)
+
     &&
-    #if $output_format == '0':
-        mv ./input.mgf '$output'
-    #else:
-        mv ./input.mzML '$output'
+
+    #if len($input) == 1:
+        #if $gzip_boolean == "0":
+            #if $output_format == "0":
+                mv ./output_folder/input.mgf ./output_file.out &&
+            #else if $output_format == "1":
+                mv ./output_folder/input.mzML ./output_file.out &&
+            #else if $output_format == "2":
+                mv ./output_folder/input.mzML ./output_file.out &&
+            ## #else if $output_format == "3":
+            ##    mv ./output_folder/input.parquet ./output_file.out &&
+            #end if
+
+        #else  ## if we have gzipped output
+            #if $output_format == "0":
```

Comment:
without gzip this would be much easier and I don't think we can use this feature effectively ... we could if we dare by enabling all datatypes with compression="true" and getting a `mzml.gz` datatype ... but not sure this is worth it now

Reply:
@CarlosHorro maybe we drop this option and make the wrapper way simpler?

Reply:
Yep I will do it :-)